packages/project/src/utils/v3-utils.ts

```diff
@@ -37,3 +51,176 @@ export const isHistoricNonCRPLandManagement = (
+                          datePlanted: crop.plantingDate,
+                          fertilizerEvents: (
+                            crop.fertilizerEvents?.map(
+                              (fertilizerEvent): V1FertilizerEvent => {
+                                return {
+                                  date: fertilizerEvent.date,
+                                  productName:
+                                    fertilizerEvent.name ||
+                                    fertilizerEvent.type,
+                                  lbsOfN: fertilizerEvent.lbsOfNPerAcre,
+                                  area: field.acres,
+                                  quantityUnit: 'lbs/acre',
+                                };
+                              }
+                            ) ?? []
+                          )
+                            .sort(
+                              (a, b) =>
+                                new Date(a.date).getTime() -
+                                new Date(b.date).getTime()
+                            )
+                            .filter(
+                              (f) =>
+                                Number(f.date.split('/').slice(-1)) ===
+                                cropYear.plantingYear // todo v3 to v1 multi year events are not supported for fertilizer events
+                            ),
+                          harvestOrKillEvents: (
+                            (crop as AnnualCrop).harvestEvents?.map(
+                              (
+                                harvestOrKillEvent: AnnualCropHarvestEvent
+                              ): V1HarvestOrKillEvent => {
+                                if (harvestOrKillEvent.yieldUnit !== 'bu/ac') {
+                                  throw new Error(
+                                    'Only bu/ac is supported as a yield unit'
+                                  );
+                                }
+                                return {
+                                  date: harvestOrKillEvent.date,
+                                  boundaryYield: harvestOrKillEvent.yield,
+                                  yieldNumeratorUnit: 'bu',
+                                  yieldDenominatorUnit: 'ac',
+                                };
+                              }
+                            ) ?? []
+                          ).sort(
+                            (a, b) =>
+                              new Date(a.date).getTime() -
+                              new Date(b.date).getTime()
+                          ),
+                          irrigationEvents:
+                            crop.irrigationEvents?.map(
+                              (irrigationEvent): V1IrrigationEvent => {
+                                return {
+                                  date: irrigationEvent.date,
+                                  volume: irrigationEvent.volume,
+                                  startDate: undefined,
```

Comment:
I *think* it's a bug in `V1IrrigationEvent` that the rest of these properties are non-nullable but we can check with Jaycen about it when he's back.